api/src/pcapi/core/providers/titelive_api.py

```diff
@@ -120,9 +120,17 @@ def filter_recent_products(
         titelive_product_page: TiteliveProductSearchResponse[TiteliveSearchResultType],
         from_date: datetime.date,
     ) -> TiteliveProductSearchResponse[TiteliveSearchResultType]:
-        """Filters out articles that were not updated since from_date."""
+        """Filters out articles that were not updated since from_date or published at from_date."""
         for oeuvre in titelive_product_page.result:
-            oeuvre.article = [article for article in oeuvre.article if article.datemodification >= from_date]
+            articles = []
+            for article in oeuvre.article:
+                if article.datemodification is None:
```

Comment:
(optionel) question de goût/lisibilité: on peux extraire la logique de is_article_recently_updated dans une fonction et garder la syntax
`oeuvre.article = [article for article in oeuvre.article if is_article_recently_updated(article)]`